### DISCUSSION TITLE
Why invalid format ed25519 key on windows

### DISCUSSION THREAD

**Discussion by kdevkr** [aws/aws-cli#7074] [Q&A] [answered]
I'm generate key-pair based ed25519 using aws-cli (ec2 console).

If i use the above key pair when EC2 is running, an invalid format error occurs when SSH is connected from a Windows.

```sh
ubuntu@ubuntu:~$ ssh -i aws-mambo.pem ec2-user@xxx
Load key "aws-mambo.pem": invalid format
ec2-user@xxx: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
```

If i create an ed25519 key pair with Windows' ssh-keygen, you can use both Windows and Linux.

I just wonder what the difference between these two is. What should I do to get the relevant information?

**Comment by aBurmeseDev** [comment, marked as answer]
Hi @kdevkr, thanks for reaching out and sorry to hear you’re getting invalid format error. Unfortunately, ED25519 keys are not supported for Windows instances as of today as mentioned [here](https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/ec2-key-pairs.html). 

> Amazon EC2 supports 2048-bit SSH-2 RSA keys for Windows instances. ED25519 keys are not supported for Windows instances.

You can also find more information about creating a key pair in different methods [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/create-key-pairs.html). Hope you find it helpful!

**Reply by kdevkr** [reply]
EC2 is based on an Amazon Linux 2 image, not a Windows instance, and only occurs when the SSH client is a Windows environment.

In the document you mentioned, Doesn't Windows Instance mean EC2?.

**Reply by aBurmeseDev** [reply]
Hi @kdevkr, I think there are a few possible fixes we could try.
One of them being the spacing in your private/public key as mentioned in this [StackOverflow comment](https://stackoverflow.com/a/68593742). As other users suggested, adding a newline at the end of the key file might be worth a try. Sometimes removing extras newlines between the start/end “headers” could solve the issue. Please let us know if that fixes the issue. 
Also, if you could share any additional information (with any sensitive info redacted), it would give me more insight into reproducing the issue on my end.

**Reply by kdevkr** [reply]
When I checked the key pair made through aws-cli in Windows, it was confirmed that it was **CRLF** with **UTF-16LE** encoding format.
After changing from CRLF to LF, there was the same problem, but I changed from UTF-16LE to UTF-8 and tried, and the connection was successful.

**Information**
- OS: Windows 10 Pro
- Build Version: 20H2 19042.746
- AWS CLI Version:  aws-cli/2.7.11 Python/3.9.11 Windows/10 exe/AMD64 prompt/off
- OpenSSH Version: OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5

Thank you for your help.
